src/utils/types.ts

```diff
@@ -146,15 +146,19 @@ export enum HttpMethod {
   DELETE = "DELETE",
 }
 
-export interface InternalRequest extends NextApiRequest {
+export type StringsDict = { [key: string]: string | string[] };
+export type UnknownDict = { [key: string]: unknown };
+
+export interface InternalRequest<Q_P, B> extends Omit<NextApiRequest, "query"> {
   user: SessionUser;
-  body: { [key: string]: unknown };
+  query: Q_P;
+  body: B;
```

Comment:
Alternatively, if we want to preserve the original body, we can do `validatedBody` and `validatedQuery`. It's possible validation will change some values depending on if we allow type coercion

Reply:
Is there a reason we would want to preserve rather than use the validated body and query?

Reply:
True. There's no real use case. Technically, validation can change types using type coercion ([AKA parse "true" to a boolean](https://ajv.js.org/coercion.html))